app/controllers/profiles_controller.rb

```diff
@@ -68,175 +63,106 @@ def render_footer?
   def set_profile
     @profile = Profile.friendly.find(params[:id])
   end
 
-  # Never trust parameters from the scary internet, only allow the white list through.
   def profile_params
     params.require(:profile).permit(
-      :email,
-      :password,
-      :password_confirmation,
-      :remember_me,
-      :country,
-      :state,
-      { iso_languages: [] },
-      :firstname,
-      :lastname,
-      :content,
-      :name,
-      :topic_list,
-      :medialinks,
-      :slug,
-      :admin_comment,
-      :main_topic_en,
-      :main_topic_de,
-      :bio_en,
-      :bio_de,
-      :twitter_de,
-      :twitter_en,
-      :website_de,
-      :website_en,
-      :website_2_de,
-      :website_2_en,
-      :website_3_de,
-      :website_3_en,
-      :profession_en,
-      :profession_de,
-      :city_de,
-      :city_en,
-      :image,
-      :copyright,
-      :personal_note_de,
-      :personal_note_en,
-      :willing_to_travel,
-      :nonprofit,
-      :inactive,
-      feature_ids: [],
-      service_ids: [],
-      translations_attributes: %i[id bio main_topic twitter website profession city locale]
+      :email, :password, :password_confirmation, :remember_me, :country, :state, { iso_languages: [] },
+      :firstname, :lastname, :content, :name, :topic_list, :medialinks, :slug, :admin_comment, :main_topic_en,
+      :main_topic_de, :bio_en, :bio_de, :twitter_de, :twitter_en, :website_de, :website_en, :website_2_de,
+      :website_2_en, :website_3_de, :website_3_en, :profession_en, :profession_de, :city_de, :city_en, :image,
+      :copyright, :personal_note_de, :personal_note_en, :willing_to_travel, :nonprofit, :inactive,
+      feature_ids: [], service_ids: [], translations_attributes: %i[id bio main_topic twitter website profession city locale]
     )
   end
 
   def search_with_search_params
-    @profiles = matching_profiles.map(&:profile_card_details)
-    @pagy, @records = pagy_array(@profiles)
-    # search results aggregated according to certain attributes to display as filters
-    aggs = ProfileGrouper.new(params[:locale], @profiles.map { |profile| profile[:id] }).agg_hash
-    @aggs_languages = aggs[:languages]
-    @aggs_cities = aggs[:cities]
-    @aggs_states = aggs[:states]
-    @aggs_countries = aggs[:countries]
-  end
-
-  def matching_profiles
-    chain ||= Profile
-      .includes(:translations)
-      .with_attached_image
-      .is_published
-      .by_region(current_region)
-      .search(params[:search])
-
-    if params[:filter_city]
-      chain = chain.by_city(params[:filter_city])
-    end
-
-    if params[:filter_country]
-      chain = chain.by_country(params[:filter_country])
-    end
-
-    if params[:filter_language]
-      chain = chain.by_language(params[:filter_language])
-    end
-
-    if params[:filter_state]
-      chain = chain.by_state(params[:filter_state])
+    result = SearchProfiles.call(params: params, region: current_region)
```

Comment:
Probably we should find a better name to distinguish the interactors better. I only can come up with `SearchProfileByAllParams` to describe better what really happens in contrast to `SearchProfileByTag`etc.